conn.go

```diff
@@ -72,3 +74,98 @@ func (c *Conn) newError(apiName string, handle interface{}) error {
 	}
 	return err
 }
+
+// QueryContext implements the driver.QueryerContext interface.
+// As per the specifications, it honours the context timeout and
+// returns when the context is cancelled.
+// When the context is cancelled, it first cancels the statement,
+// then closes it, and returns an error.
+func (c *Conn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	// Prepare a query
+	os, err := c.PrepareODBCStmt(query)
+	if err != nil {
+		return nil, err
+	}
+
+	dargs, err := namedValueToValue(args)
+	if err != nil {
+		return nil, err
+	}
+
+	// Execute the statement
+	rowsChan := make(chan driver.Rows)
+	defer close(rowsChan)
+	errorChan := make(chan error)
+	defer close(errorChan)
+
+	if ctx.Err() != nil {
+		os.closeByStmt()
+		return nil, ctx.Err()
+	}
+
+	runQuery := func() {
+		err := os.Exec(dargs, c)
+		if err != nil {
+			errorChan <- err
+			return
+		}
+
+		err = os.BindColumns()
+		if err != nil {
+			errorChan <- err
+			return
+		}
+
+		os.usedByRows = true
+		rowsChan <- &Rows{os: os}
+
+		// At the end of the execution, we check if the context has been cancelled
+		// to ensure there's no race condition below (L144).
+		if ctx.Err() != nil {
+			errorChan <- err
+		}
+	}
+
+	go runQuery()
+
+	var finalErr error
+	var finalRes driver.Rows
+
+	select {
+	case <-ctx.Done():
+		err := os.Cancel()
+		if err != nil {
```

Comment:
and ditto a few other places. If you too like this pattern, that is!

```suggestion
		if err := os.Cancel(); err != nil {
```

Reply:
But this is the magic sauce here, right? The fact that we call `Cancel`.

Reply:
This is correct; I've added a comment to hopefully make it clearer